DatamartManagementService/DatamartManagementService.Domain/RevenueSummaryRetrievalService.cs

```diff
@@ -0,0 +1,33 @@
+﻿using DatamartManagementService.Domain.Mappers.Database;
+using DatamartManagementService.Domain.Models.RofDatamartModels;
+using DatamartManagementService.Infrastructure.Persistence.RofDatamartRepos;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace DatamartManagementService.Domain
+{
+    public interface IRevenueSummaryRetrievalService
+    {
+        Task<List<RofRevenueByDate>> GetRevenueBetweenDates(DateTime startDate, DateTime endDate);
+    }
+
+    public class RevenueSummaryRetrievalService : IRevenueSummaryRetrievalService
+    {
+        private readonly IRevenueByDateRetrievalRepository _revenueByDateRetrievalRepo;
+
+        public RevenueSummaryRetrievalService(IRevenueByDateRetrievalRepository revenueByDateRetrievalRepo)
+        {
+            _revenueByDateRetrievalRepo = revenueByDateRetrievalRepo;
+        }
+
+        public async Task<List<RofRevenueByDate>> GetRevenueBetweenDates(DateTime startDate, DateTime endDate)

```

Comment:
Not just RevenueBetweenDates.
After getting all revenue between the dates, i want you to group them, remember?
Display a count of each one and total revenue and gross revenue.

Reply:
added